### DISCUSSION TITLE
Does --theme=ansi improve performance?

### DISCUSSION THREAD

**Discussion by linrongbin16** [sharkdp/bat#2451] [Q&A] [answered]
Does --theme=ansi improve bat performance?

**Comment by keith-hall** [comment, marked as answer]
I personally wouldn't expect it to make much difference, the same syntax highlighting engine is used, the same color scheme processing rules, just presumably with less colors. You can always benchmark it though.

**Reply by linrongbin16** [reply]
thanks

**Comment by sharkdp** [comment]
Well. Let's measure it:
```bash
hyperfine \
  --parameter-list theme default,ansi \
  --export-markdown=theme.md \
  'bat -f --theme={theme} jquery.js'
```
(I'm using the file `tests/benchmarks/highlighting-speed-src/jquery.js` from this repo)

Yes. `bat` is 25% faster when using the `ansi` theme:

| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `bat -f --theme=default jquery.js` | 655.7 ± 4.4 | 649.2 | 660.8 | 1.31 ± 0.01 |
| `bat -f --theme=ansi jquery.js` | 501.6 ± 3.3 | 498.1 | 509.2 | 1.00 |

That might come down to having less output to write(?). Because the ANSI escape sequences for a 24bit-color-theme are much longer.

I also ran a benchmark using `hyperfine`s `--show-output` option, which measures the performance including the terminals rendering speed, because I suspected that might make a difference. But `bat` seems to be the limiting factor here. The results are pretty much the same as above.

**Reply by linrongbin16** [reply]
really thank for your benchmark!
